lineage/states/StateDistribution.py

```diff
@@ -87,25 +97,27 @@ def __repr__(self):
 # can handle the case where the list of observations is empty.
 
 
-def gamma_estimator(gamma_obs):
+def gamma_estimator(gamma_obs, gamma_censor_obs, old_params, gammas):
     """
     This is a closed-form estimator for two parameters
     of the Gamma distribution, which is corrected for bias.
     """
-    N = len(gamma_obs)
+    s = np.log(sum(gammas * gamma_obs) / sum(gammas)) - sum(gammas * np.log(gamma_obs)) / sum(gammas)
+    k0 = old_params[0]
+
+    def f(k, s): return 1/(np.log(k) - sc.polygamma(0, k)) - 1/s
+    def fprime(k, s): return (k*sc.polygamma(1, k) - 1.)/(k*(np.log(k)-sc.polygamma(0, k))**2)
+    def fprime2(k, s): return (2*((1/k)-sc.polygamma(1, k))**2)/((np.log(k)-sc.polygamma(0, k))**3) + (sc.polygamma(2, k)+(1./k**2))/((np.log(k)-sc.polygamma(0, k))**2)
 
-    xbar = (sum(gamma_obs) + 49e-10) / (len(gamma_obs) + 1e-10)
-    x_lnx = [x * np.log(x) for x in gamma_obs]
-    lnx = [np.log(x) for x in gamma_obs]
-    # gamma_a
-    a_hat = (N * (sum(gamma_obs)) + 7e-10) / (N * sum(x_lnx) - (sum(lnx)) * (sum(gamma_obs)) + 1e-10)
-    # gamma_scale
-    b_hat = xbar / a_hat
+    def halley(k, s): return k - ((2 * (f(k, s) * fprime(k, s))) / (2 * (fprime(k, s)**2) - f(k, s) * fprime2(k, s)))
```

Comment:
Isn't this only doing one iteration of Halley?

Reply:
Yep but I was getting it to the answer within one to two iterations so I wasn't sure if a loop was necessary.

Reply:
The second iteration was always a very small number away from the first iteration (around ~0.0005)

Reply:
Really think you still need to iterate... there could easily be pathological cases.

Reply:
If you're having trouble with this I'm fine reverting to a version that's working (one iteration) and then just opening an issue to revisit this later.

Reply:
Yeah I can't seem to figure it out, and I like approaching the problem on a new issue/branch later.